GATEWAY/src/providers/devices.service.ts

```diff
@@ -7,6 +7,8 @@ import { Device } from './utils.service';
 @Injectable()
 export class DevicesService {
   public devices: Device[];
+  public flagThen: boolean = false;
+  public flagCatch: boolean = false;
```

Comment:
that is because the only difference is if the object was stored in the app storage or not. You can test it by adding an object to the storage and changing its name propery and then see which object is returned.